plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -1130,24 +1128,24 @@ public Set<HivePrivilegeInfo> listTablePrivileges(String databaseName, String ta
             implements AsyncHandler<Request, Result>
     {
         private final GlueMetastoreApiStats stats;
-        private final long startTimeInMillis;
+        private final long startTimeNanos;
 
-        public StatsRecordingAsyncHandler(GlueMetastoreApiStats stats, long startTimeInMillis)
+        public StatsRecordingAsyncHandler(GlueMetastoreApiStats stats)
         {
             this.stats = requireNonNull(stats, "stats is null");
-            this.startTimeInMillis = startTimeInMillis;
+            this.startTimeNanos = System.nanoTime();
         }
 
         @Override
         public void onError(Exception e)
         {
-            stats.recordCall(System.currentTimeMillis() - startTimeInMillis, true);
+            stats.recordCall(NANOSECONDS.toMillis(System.nanoTime() - startTimeNanos), true);
```

Comment:
This would be so more readable:

```suggestion
            stats.recordCall(stopwatch.elapsed(MILLISECONDS), true);
```

Reply:
Changed